### ISSUE TITLE
Group.name property returns full path instead of name

### ISSUE THREAD

**Issue by kmuehlbauer** [h5netcdf/h5netcdf#68]
XRef:  https://github.com/pydata/xarray/issues/3680 for more detail.

Sorry, if that has been discussed before.

I see two points here:

- alignment with h5py which returns the full name of the object (full path from root), status quo
- alignment with netCDF4 which returns the name

Xarray tries to use `h5netcdf` in the same manner as `netCDF4`, which breaks due to this.

The questions is, where should this be fixed? Thanks for any ideas!

**Comment by kmuehlbauer** [comment]
Time to find a solution to this issue. 

```python
import netCDF4
import h5netcdf.legacyapi as h5nc4
with netCDF4.Dataset("testx.nc", "w", format="NETCDF4") as root:
    test1 = root.createGroup("test1")
    test1.createGroup("test2")
    print(root['/test1/test2'].name)
with h5nc4.Dataset('testx.nc') as h5n:
    print(h5n["/test1/test2"].name)

test2
/test1/test2
```

So, at least for the legacy API we should try to conform to netCDF4. This will not help https://github.com/pydata/xarray/issues/3680, but I do see no way to resolve this in another way (besides breaking conformity with `h5py`). Do you have any preferences, @shoyer?

**Comment by kmuehlbauer** [comment]
resolved in #126
